doc/chapters/debugging/index.rst

```diff
@@ -0,0 +1,197 @@
+
+*lxc-info* provides information about the running container, such as network
+interface used for t he bridge, the state of the container, what PID it has and
+so forth.
+
+Cleanup
+-------
+
+The internal filesystem will always be deleted and recreated when an app is
+restarted. This means that anything you want to save during debugging needs to
+be saved to a bind mounted directory before you shut down the container.
+
+`SoftwareContainer` will also delete all container instances upon startup,
+meaning that container instances will be lost when restarting.
+
+GDB
+---
+
+Attaching a `gdb` session can be done in two ways, one is to run the `gdb` session
+and attach inside the container, the other is to attach to the process from the
+outside of the container. Using both methods have their own set of problems you
+need to solve.
+
+Running GDB inside container
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you run `gdb` from inside the container using `lxc-attach`, you should not
+need to change any paths etc to find the correct libraries etc. You will need
+to have the `PID` from inside the container available to connect to however. this
+can be found out by running::
+
+    sudo lxc-attach --name SC-X ps aux
+
+Then running::
+
+    sudo lxc-attach --name SC-X -- gdb --pid YY
+
+Replace YY with the `PID` as seen inside the container. You can also run the
+application directly inside the container and start it with gdb directly::
+
+    sudo lxc-attach --name SC-X -- gdb --args /bin/ls
+
+TODO: Uncertain why, but this fails with unable to find executable.
+
+Running GDB outside container
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you run `gdb` from outside the container you will run into some problems with the
+paths as the application is executing inside a `chroot` and `gdb` is executing
+outside of it, this is fairly easily fixed however. First off you need to
+provide `gdb` with the path of the executable as the path provided by the system
+is inside the chroot::
+
+    # ps aux |grep main
+    # sudo gdb --pid XXX /tmp/container//SC-0/gateways/app/main
+
+Begin with finding the global namespace `PID` of the application you are
+debugging and then attach `gdb` to that `PID`, and the path to the application
+in the `un-chrooted` filesystem.
+
+The application should now be possible to debug as normal. Most of the standard
+libraries etc are mounted inside the `chroot` from the outside and should be
+available as normal without any changes. If you have dynamic libraries inside
+your container that the application are linking to, you may need to add the
+proper `solib-search-paths` to the libraries.
+
+Debugging using QtCreator inside a container
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Debugging applications running inside a container can be done, however, all
+processes are run as root so you need to start `QtCreator` as root to be
+able to connect to the process inside the container.
```

Comment:
I dont think so, but I haven't looked into ways around it really. It's part of the security model of Linux imho, and being able to debug a root app from a non-root app would be strongly discouraged imho.